lib/BaseClient.ts

```diff
@@ -45,6 +45,13 @@ abstract class BaseClient extends EventEmitter {
   public isDisconnected(): boolean {
     return this.status === ClientStatus.Disconnected;
   }
+  public async close() {
+    if (this.reconnectionTimer) {
+      clearTimeout(this.reconnectionTimer);
+    }
+    await this.stop();
+  }
+  protected abstract stop(): Promise<void>;
```

Comment:
Why have this return a promise if we're not doing anything asynchronous with it?

Reply:
Another thought I have here is that "close" and "stop" might get confused with each other, maybe it'd be more descriptive to call this method "closeSpecific" or something like that?

Reply:
> Why have this return a promise if we're not doing anything asynchronous with it?
I was assuming we're going to have async cleanup in the future, but you're right - it's not necessary right now. Removed it.

> maybe it'd be more descriptive to call this method "closeSpecific"
`closeSpecific` sounds great. Changed.